core/domain/state_domain.py

```diff
@@ -826,59 +829,77 @@ def validate(self):
             ValidationError: One or more attributes of the ContentTranslations
             are invalid.
         """
-        for (content_id, language_code_to_translation_script) in (
+        for (content_id, language_code_to_written_translation) in (
                 self.content_translations.iteritems()):
             if not isinstance(content_id, basestring):
                 raise utils.ValidationError(
                     'Expected content_id to be a string, received %s'
                     % content_id)
-            if not isinstance(language_code_to_translation_script, dict):
+            if not isinstance(language_code_to_written_translation, dict):
                 raise utils.ValidationError(
                     'Expected content_id value to be a dict, received %s'
-                    % language_code_to_translation_script)
-            for (language_code, translation_script) in (
-                    language_code_to_translation_script.iteritems()):
+                    % language_code_to_written_translation)
+            for (language_code, written_translation) in (
+                    language_code_to_written_translation.iteritems()):
                 if not isinstance(language_code, basestring):
                     raise utils.ValidationError(
                         'Expected language_code to be a string, received %s'
                         % language_code)
-                translation_script.validate()
+                if not utils.is_valid_language_code(language_code):
+                    raise utils.ValidationError(
+                        'Invalid language_code: %s' % language_code)
+
+                written_translation.validate()
 
     def get_available_languages(self):
         """Returns a set of language available in the ContentTranslations.
 
         Returns:
-            set. A set of languages available in the ContentTranslations.
+            set(str.). A set of languages available in the ContentTranslations.
```

Comment:
set(str) not set(str.)

(Ditto elsewhere.)

Reply:
Done!